kubernetes/contexts.yaml

```diff
@@ -39,8 +39,19 @@ contexts:
           image: google/cloud-sdk:latest
           command_prefix: []
           shell_entry: [ "/bin/bash", "-c" ]
+        berglas:
+          image: us-docker.pkg.dev/berglas/berglas/berglas:latest
+          command_prefix: []
+          shell_entry: [ "/bin/sh", "-c" ]
 
       predefined_steps:
+        install-berglas:
+          name: install-berglas
+          type: berglas
+          shell: |
+            mkdir -p /honeydipper/.bin
+            cp /bin/berglas /honeydipper/.bin/berglas
```

Comment:
can this be solved using `$PATH` instead?

Reply:
This installation step is to ensure the Berglas binary is available to the subsequent `containers` and `initContainers`. Without copying the binary to the shared volume, it is not possible. Adding it to the `$PATH` doesn't really help.

Reply:
How is honeydipper/.bin a shared volume here?

Reply:
> How is honeydipper/.bin a shared volume here?
> 

Yes, `/honeydipper` is an `emptyDir` mounted in every containers/initContainers of the pod.

> Would it work to use `initContainer` to copy it into an `emptyDir` volume or do `volumeMount` from `hostPath`, which I’m pretty sure you can do. That said, I’m wondering if it might make sense to either use the file method, or require that the user have a container that has the tools they need. Otherwise, it seems like maybe this is overly specific for use in config-essentials?

I am not sure what you mean by `file method`. I see plenty examples of containers acquiring binaries through curl, github clone, or modules from registries, such as in CI jobs. I can choose to do a curl download as the Berglas document suggested, which might look more conventional, but I think the copy method requires less dependencies and adds an additional layer of control on the source (we can build our own berglas image).

As for requiring the images to have the tools they need, it is already supported, just don't set the `use_berglas` flag. I would like to keep this an option for the people who prefer to do less customization, who focus more on the result instead of the solution.

Reply:
using `initContainer` would be a file copy as well.